### PR TITLE
PB-3406 : Add project mapping for rancher

### PR DESCRIPTION
**Pull request by px-kesavan**
Signed-off-by: Kesavan Thiruvenkadasamy <kthiruvenkadasamy@purestorage.com>

**What this PR does / why we need it**:

**Which issue(s) this PR fixes** (optional)
Closes #

**Special notes for your reviewer**:

### CODE REVIEW

File: pkg/apis/v1/api.proto [hunk 1]
```diff
@@ -449,6 +449,8 @@ message BackupInfo {
     BackupShare backup_share = 28;
     BackupShare.AccessType user_backupshare_access = 29;
     ObjectRef cluster_ref = 30;
+    // Name of projects associated with resources
+    repeated string projects = 31;
```
Comment (siva-portworx) [inline pkg/apis/v1/api.proto:453]: Does it make sense to have mention about rancher in the comment? So the use of the APIs will know it is specific to rancher project and not something generic.

File: pkg/apis/v1/api.proto [hunk 2]
```diff
@@ -1565,6 +1568,8 @@ message RestoreCreateRequest {
     map<string, string> storage_class_mapping = 8;
     // backup object reference having name and id
     ObjectRef backup_ref = 9;
+    // mapping of the projects from the backup to which restore should be done
```
Comment (siva-portworx) [inline pkg/apis/v1/api.proto:1571]: I feel it is better change the variable name also to include rancher in it.

Comment (siva-portworx) [reply pkg/apis/v1/api.proto:1571]: We might need to have a parameter in RestoreInfo def to store the project mapping details.

**Review by siva-portworx — APPROVED** [review]

File: pkg/apis/v1/api.proto (hunk 1 above)
Comment (px-kesavan) [reply pkg/apis/v1/api.proto]: Updated the comment and modified the variable to rancher_projects

File: pkg/apis/v1/api.proto (hunk 2 above)
Comment (px-kesavan) [reply pkg/apis/v1/api.proto]: fixed,
modified variable name to rancher_project_mapping
and added to RestoreInfo

**Pull request merged**